### PR TITLE
Store Orders: Switch to edit layout for new order screen

### PR DESCRIPTION
**Pull request by ryelle**
This PR deletes the existing, unused order creation components, and adds new components for creation state. The goal here was to get something functional rendering with the edit components, this is just a step towards the final product. With this groundwork we can make smaller PRs for following work.

When you click New Order now, this is the screen you end up on. It looks similar to the edit screen, with a slight difference in the customer info section.

empty

Once some products, fees, and addresses are filled in, it looks more like editing an existing order:

filled-out

There is no activity log component on this page (nothing has happened yet to show up) – this could leave room for us to add a more create-specific cards (See #15679 for discussion)

cc @kellychoffman @jameskoster to let you know this is starting 🙂 

**To test**

- View `/store/order/:site`
- You should see a screen that looks like the edit screen
- Try adding products, fees, change shipping cost – this should update the total cost
- Add addresses, these should display correctly on the page
- **Saving does not currently work!** So just test that interactions are as expected & no errors crop up.

### CODE REVIEW

File: client/extensions/woocommerce/app/order/create.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			<Main className={ className } wideLayout>
+				<ActionHeader breadcrumbs={ breadcrumbs }>
+					<Button
+						key="save"
+						primary
+						onClick={ this.saveOrder }
+						busy={ isSaving }
+						disabled={ ! hasOrderEdits }
+					>
+						{ translate( 'Save Order' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/order/create.js]: Hi! I've found a possible matching string that has already been translated **26** times:
`translate( 'Save New Order' )` *ES Score: 8*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-new-order-creation)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/order/order-customer/create.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+	renderShipping = ( address = {} ) => {
+		const { translate } = this.props;
+		if ( every( address, isEmpty ) ) {
+			return (
+				<Button
+					className="order-customer__add-link"
+					onClick={ this.toggleDialog( 'shipping' ) }
+					primary
+				>
+					{ translate( 'Add Shipping Address' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/order/order-customer/create.js]: Hi! I've found a possible matching string that has already been translated **20** times:
`translate( 'Shipping address' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/store-new-order-creation)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/order/order-customer/create.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	renderBilling = ( address = {} ) => {
+		const { translate } = this.props;
+		if ( every( address, isEmpty ) ) {
+			return (
+				<Button
+					className="order-customer__add-link"
+					onClick={ this.toggleDialog( 'billing' ) }
+					primary
+				>
+					{ translate( 'Add Billing Address' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/order/order-customer/create.js]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Billing Address' )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by ryelle** [comment]
Switching to in-progress to fix the broken test

**Comment by allendav** [comment]
Add fee retains the focus / outline after adding a fee - not sure if you intended that

screen shot 2017-12-11 at 9 57 32 am

File: client/extensions/woocommerce/app/order/create.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	};
+
+	render() {
+		const { className, hasOrderEdits, isSaving, orderId, site, translate } = this.props;
+		if ( ! orderId ) {
+			return null;
+		}
+
+		const breadcrumbs = [
+			<a href={ getLink( '/store/orders/:site/', site ) }>{ translate( 'Orders' ) }</a>,
```
Comment (allendav) [inline client/extensions/woocommerce/app/order/create.js]: Maybe add explanation for translators?

File: client/extensions/woocommerce/app/order/order-customer/create.js [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+	};
+
+	state = {
+		showDialog: false,
+	};
+
+	updateAddress = ( type = 'billing' ) => {
+		const { siteId, order } = this.props;
+		return address => {
+			const { copyToShipping = false, ...newAddress } = address;
```
Comment (allendav) [inline client/extensions/woocommerce/app/order/order-customer/create.js]: This new fancy JS stuff - always have to sit and think these through :)

File: client/extensions/woocommerce/app/order/order-customer/create.js [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+		}
+		return (
+			<Fragment>
+				<h4>{ translate( 'Address' ) }</h4>
+				<div className="order-customer__billing-address">
+					<p>{ `${ address.first_name } ${ address.last_name }` }</p>
+					<AddressView address={ getAddressViewFormat( address ) } />
+				</div>
+
+				<h4>{ translate( 'Email' ) }</h4>
```
Comment (allendav) [inline client/extensions/woocommerce/app/order/order-customer/create.js]: Maybe hint for translators

File: client/extensions/woocommerce/state/ui/orders/selectors.js [hunk 7]
```diff
@@ -51,6 +52,35 @@ export const getOrdersCurrentSearch = ( state, siteId = getSelectedSiteId( state
...
+ * Get a default order "frame", so we have values for components.
+ * @return {Object} The local edits made to the current order
+ */
+export const getDefaultEmptyOrder = () => {
+	const currency = 'USD';
+
+	return {
+		status: 'pending',
+		currency: currency,
+		discount_total: getCurrencyFormatString( 0, currency ),
```
Comment (allendav) [inline client/extensions/woocommerce/state/ui/orders/selectors.js]: Maybe DRY these calls into a constant?

**Comment by kellychoffman** [comment]
Nice! 

One request: Can `+ Add Product` link be a Primary Button instead? This way, it will align with the `Add Billing Address` and  `Add Shipping Address` Primary Buttons, and together it will be immediately obvious which things you should add to create an order. (Fees seem low used and optional, so I think we should keep that link as is)

**Comment by ryelle** [comment]
@kellychoffman I've updated the button to primary, but this component is used on both the new & existing edit screens:

new

existing

IMO it's fine to change in both, since just the one button is not overpowering, but I want to get a 👍  before I merge.

**Comment by kellychoffman** [comment]
Could I be a pain and ask you to switch the order so that `+ Add Fee` comes before `+Add Product`? Other than that, good to 🚢 .

**Pull request merged**